src/main/kotlin/no/nav/familie/ef/iverksett/brev/DistribuerVedtaksbrevTask.kt

```diff
@@ -17,17 +18,47 @@ import java.util.UUID
                      triggerTidVedFeilISekunder = 15 * 60L,
                      beskrivelse = "Distribuerer vedtaksbrev.")
 class DistribuerVedtaksbrevTask(private val journalpostClient: JournalpostClient,
-                                private val tilstandRepository: TilstandRepository) : AsyncTaskStep {
+                                private val tilstandRepository: TilstandRepository,
+                                private val featureToggleService: FeatureToggleService) : AsyncTaskStep {
 
     private val logger: Logger = LoggerFactory.getLogger(this::class.java)
 
     override fun doTask(task: Task) {
         val behandlingId = UUID.fromString(task.payload)
+
         val journalpostId = tilstandRepository.hentJournalpostResultat(behandlingId)?.journalpostId
-        val bestillingId = journalpostId?.let { journalpostClient.distribuerBrev(it) }
-        tilstandRepository.oppdaterDistribuerVedtaksbrevResultat(behandlingId = behandlingId,
-                                                                 DistribuerVedtaksbrevResultat(bestillingId = bestillingId)
-        )
+        journalpostId
+                ?.let { journalpostClient.distribuerBrev(it) }
+                ?.also { bestillingId ->
+                    loggBrevDistribuert(journalpostId, behandlingId, bestillingId)
+                    tilstandRepository.oppdaterDistribuerVedtaksbrevResultat(behandlingId = behandlingId,
+                                                                             DistribuerVedtaksbrevResultat(bestillingId = bestillingId)
```

Comment:
Skal dette skje dersom vi skal distribuere til brevmottakere? Hvis ikke bør vel kanskje dette inn i en `if(toggle==true){ distribuerforbrevmottakere } else { distribuerPåGamleMåten }`

Reply:
distribuerforbrevmottakere vil kun kjøres for dem som dem som har satt brevmottakere, så "den gamle" koden skal fortsatt kjøre for 99% av casene. Skal forhåpentligvis aldri komme i en situasjon der begge vil trigges, men kan ta validering på dette også.

Reply:
Men kjører ikke den gamle koden alltid nå, eller er det jeg som er blind? :)